### DISCUSSION TITLE
How to crossref a LaTeX figure env picture?

### DISCUSSION THREAD

**Discussion by produnis** [quarto-dev/quarto-cli#8960] [Q&A] [answered]
### Description

Hi there, my quarto theme uses scrartcl in twocolumn mode.
In order to insert a picture, that fits over both columns, I use:

````
```{=latex}
\begin{figure*}
\centering
\includegraphics[]{my_wide_pic.png}
\caption{Zweinspaltige Breite mittels figure*}
\label{fig-zweispaltig}
\end{figure*}
```
````

This looks just fine, however, I am not able to crossref this pic with `@fig-zweispaltig`.

If I change `\label{fig-zweispaltig}` to `\label{tbl-zweispaltig}` and `@tbl-zweispaltig`, it works, but then it's no longer a Picture 1.13, but a Table 1.13.

So,  how can I crossref my LaTeX-figure-env picture?

thx in advance

**Comment by mcanouil** [comment, marked as answer]
What's the purpose exactly of using LaTeX? (Note that your example is not reproducible.)

Here you basically are saying "I don't want Quarto to handle cross-reference, I'll handle it myself" (at least for your not reproducible example).

````qmd
---
format: pdf
---

::: {#fig-zweispaltig fig-env="figure*"}

![](https://placehold.co/600x400.png)

Zweinspaltige Breite mittels figure

:::
````

**Reply by produnis** [reply]
Thx a lot, I was not aware that I could Set the figure* env that way!
Works just fine and solves my needs

Thx again!

**Reply by mcanouil** [reply]
Note that you can also use:

```
![Zweinspaltige Breite mittels figure](https://placehold.co/600x400.png){#fig-zweispaltig fig-env="figure*"}
```

See <https://quarto.org/docs/reference/formats/pdf.html#figures>.
Note that the search for this case is really straightforward:
<img width="889" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/d51d85f9-de89-430d-bc4b-5259235bdd01">

**Reply by produnis** [reply]
I had this issue with longtable in twocolumn, (<https://github.com/quarto-dev/quarto-cli/issues/7236>) and there, the workaround is with raw latex code. It works as well with table* to use full width in twocolum Mode.
And as I moved on to Figures, I just adapted the tbl approach without checking the figure refs carefully.

Thx for helping me!

**Reply by mcanouil** [reply]
About longtable (as bad as it is), there were some issues with it and some fix/workaround have been implemented.
Take a look at the issues.

**Reply by produnis** [reply]
yes, I cam from here <https://github.com/quarto-dev/quarto-cli/issues/7236#issuecomment-1764782266> and just adopted it to figures without thinking... ;)
